cpp/include/cuml/tree/flatnode.h

```diff
@@ -35,6 +35,16 @@ struct SparseTreeNode {
   uint32_t instance_count = UINT32_MAX;  // UINT32_MAX indicates n/a
 };
 
+template <typename DataT, typename LabelT, typename IdxT>
+bool operator==(const SparseTreeNode<DataT, LabelT, IdxT>& lhs,
+                const SparseTreeNode<DataT, LabelT, IdxT>& rhs)
+{
+  return (lhs.prediction == rhs.prediction) && (lhs.colid == rhs.colid) &&
```

Comment:
Do we expect the floating points values such as `prediction` etc, match exactly?

Reply:
I've disabled any checks like this for now, but in the future I don't think that's an unreasonable goal for classification.